apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSink.java

```diff
@@ -49,33 +70,84 @@ public void close() throws Exception {
     }
   }
 
-  public int writeBatch(Collection<RdaChange<TClaim>> allClaims) throws ProcessingException {
-    final Collection<RdaChange<TClaim>> claims = dedupChanges(allClaims);
+  /**
+   * Queries the RdaApiProgress table to get the maximum sequence number for our claim type.
+   *
+   * @return max sequence number or empty if there is no record for this claim type
+   * @throws ProcessingException if the operation fails
+   */
+  @Override
+  public Optional<Long> readMaxExistingSequenceNumber() throws ProcessingException {
+    try {
+      logger.info("running query to find max sequence number");
+      String query =
+          String.format(
+              "select p.%s from RdaApiProgress p where p.claimType='%s'",
+              RdaApiProgress.Fields.lastSequenceNumber, claimType.name());
+      final List<Long> sequenceNumber =
+          entityManager.createQuery(query, Long.class).getResultList();
+      final Optional<Long> answer =
+          sequenceNumber.isEmpty() ? Optional.empty() : Optional.of(sequenceNumber.get(0));
+      logger.info(
+          "max sequence number result is {}", answer.map(n -> Long.toString(n)).orElse("none"));
```

Comment:
This seems like it could possibly be a non-trivial calculation, should we wrap this with a `logger.infoEnabled()` check?

Reply:
Since answer is an optional this isn't really expensive.